models/classes/Translation/Service/QtiIdentifierRetriever.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2024 (original work) Open Assessment Technologies SA.
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiTest\models\Translation\Service;
+
+use core_kernel_classes_Resource;
+use Psr\Log\LoggerInterface;
+use taoQtiTest_models_classes_QtiTestService;
+
+class QtiIdentifierRetriever
+{
+    private taoQtiTest_models_classes_QtiTestService $qtiTestService;
+    private LoggerInterface $logger;
+
+    public function __construct(taoQtiTest_models_classes_QtiTestService $qtiTestService, LoggerInterface $logger)
+    {
+        $this->qtiTestService = $qtiTestService;
+        $this->logger = $logger;
+    }
+
+    public function retrieve(core_kernel_classes_Resource $test): ?string
+    {
+        try {
+            $jsonTest = $this->qtiTestService->getJsonTest($test);
+        } catch (Throwable $exception) {
+            $this->logger->error('An error occurred while retrieving test data: ' . $exception->getMessage());
+
+            return null;
+        }
+
+        return json_decode($jsonTest, true)['identifier'] ?? null;
```

Comment:
@shpran I think the code can be simplified. I am also not sure we should silence that exception. If the identifier is not created, there is something wrong.

```suggestion
        try {            
            return json_decode($this->qtiTestService->getJsonTest($test), true)['identifier'] ?? null;
        } catch (Throwable $exception) {
            $this->logger->error('An error occurred while retrieving test data: ' . $exception->getMessage());

            throw $e;
        }
```

Reply:
Done ✅